src/lib/operators/export_csv.cpp

```diff
@@ -21,31 +22,30 @@ ExportCsv::ExportCsv(const std::shared_ptr<const AbstractOperator> in, const std
 const std::string ExportCsv::name() const { return "ExportCSV"; }
 
 std::shared_ptr<const Table> ExportCsv::_on_execute() {
-  CsvConfig config{};
-  _generate_meta_info_file(_input_left->get_output(), _filename + config.meta_file_extension);
+  _generate_meta_info_file(_input_left->get_output(), _filename + CsvMeta::META_FILE_EXTENSION);
   _generate_content_file(_input_left->get_output(), _filename);
 
   return _input_left->get_output();
 }
 
-void ExportCsv::_generate_meta_info_file(const std::shared_ptr<const Table>& table, const std::string& meta_file) {
-  CsvWriter writer(meta_file);
-  // Write header line
-  writer.write_line({"PropertyType", "Key", "Value"});
-
-  // Chunk size
-  writer.write_line({"ChunkSize", "", static_cast<int64_t>(table->chunk_size())});
+void ExportCsv::_generate_meta_info_file(const std::shared_ptr<const Table>& table, const std::string& meta_file_path) {
+  CsvMeta meta{};
+  meta.chunk_size = table->chunk_size();
 
   // Column Types
   for (ColumnID col_id{0}; col_id < table->column_count(); ++col_id) {
-    auto type = table->column_type(col_id);
-
-    if (table->column_is_nullable(col_id)) {
-      type += std::string("_") + CsvConfig::NULL_STRING;
-    }
+    ColumnMeta column_meta;
+    column_meta.name = table->column_name(col_id);
+    column_meta.type = table->column_type(col_id);
+    column_meta.nullable = table->column_is_nullable(col_id);
 
-    writer.write_line({"ColumnType", table->column_name(col_id), type});
+    meta.columns.push_back(column_meta);
   }
+
+  nlohmann::json meta_json = meta;
```

Comment:
It would be cleaner to use the constructor explicitly instead of relying on the implicit conversion.

Reply:
With using constructor explicitly, do you mean something like `auto meta_json = nlohmann::json(meta);`? Otherwise, as stated above, we basically rely in the conversion that we defined ourselves, that was what the `to_json` and `from_json` was for.

Reply:
Yes. I feel like assignments should only be used in places where the left and right sides are somewhat the same. I'm not a big fan of implicit conversions. Don't feel strongly enough about it to block the PR for this.